ksqldb-serde/src/main/java/io/confluent/ksql/serde/protobuf/ProtobufSerdeFactory.java

```diff
@@ -67,15 +115,29 @@ <T> Serde<T> createSerde(
       final boolean isKey) {
     validate(schema);
 
-    final Optional<Schema> physicalSchema = properties.getSchemaId().isPresent() ? Optional.of(
-        SerdeUtils.getAndTranslateSchema(srFactory, properties.getSchemaId()
-            .get(), new ProtobufSchemaTranslator(properties))) : Optional.empty();
+    // If a SR schema exists (physical schema), then fetch it and use it for the serializer.
+    // The schema ID is fetched along the physical schema to be used in case multiple schema
+    // definitions are found, which will define how Connect serializer will work (see
+    // the getConverter() class in this file)
+    final Optional<SchemaAndId> schemaAndId = getSchemaFromSR(srFactory);
+    final Optional<Schema> physicalSchema = schemaAndId.map(SchemaAndId::toConnectSchema);
+
+    // The schemaId may be used when creating the ProtobufConverter for the serializer.
+    // It should be used only if SCHEMA_ID was specified manually or if the SR schema has
+    // multiple schema definitions.
```

Comment:
What's the reason for using schema ID if the schema has multiple schema definitions? Is it because each schema definition within the schema has its own schema ID?

Reply:
This is only to activate the config used in the converter. The config I mentioned here (https://github.com/confluentinc/ksql/pull/9038#discussion_r853032771). I need to disable the auto.register.schemas and also make the SR API take the path when the schema id is present. 

I need a better comment.